File: Php/Infrastructure/Data/MySql/MySqlTwitterStorage.php
namespace Infrastructure\Data\MySql {

	use Infrastructure\Data\Interfaces\TwitterStorage;
	use Infrastructure\Data\MySql\StorageProvider;
	use Domain\Classes\User;
	use integration\Sharing\TwitterCredentials;

	class MySqlTwitterStorage implements TwitterStorage{

		private $db;

		public function __construct(StorageProvider $db){
			db = $db;
		}

		public function getCredentials(User $user){
			$sql =	"SELECT ut.UserID, ut.TwitterName, ut.Key, ut.Secret " .
					"FROM usertwitter ut " .
					"WHERE ut.UserID = {$user.getId()}";
			return getCredentialsFromSql($sql);
		}

		public function addCredentials(User $user, TwitterCredentials $credentials){
			$sql =	"INSERT INTO usertwitter " .
					"(UserID, TwitterName, `Key`, Secret) " .
					"VALUES " .
					"({$user.getId()}, '{$credentials.key}', '{$credentials.secret}')";
			$rowCount = db.execute($sql);
			return db.getLatestInsertId($rowCount > 0);
		}

		public function clearCredentials(User $user){
			$sql =	"DELETE FROM usertwitter " .
					"WHERE UserID = {$user.getId()}";
			$rowCount = db.execute($sql);
			return $rowCount > 0;
		}

		private function getCredentialsFromSql($sql){
			$res = db.query($sql);
			$player = null;
			foreach($res.fetchAll() as $row){
				$player = credentialsFromDbRow($row);
				break;
			}
			return $player;
		}

		private function credentialsFromDbRow($row){
			$credentials = new TwitterCredentials();
			$credentials.twitterName = $row["TwitterName"];
			$credentials.key = $row["Key"];
			$credentials.secret = $row["Secret"];
			return $credentials;
		}

	}

}

File: Infrastructure/Data/Storage/MySqlTwitterStorage.cs
using Infrastructure.Data.Classes;
using Infrastructure.Data.Factories;
using Infrastructure.Data.Storage.Interfaces;

namespace Infrastructure.Data.Storage {

	public class MySqlTwitterStorage : ITwitterStorage
    {
	    private readonly IStorageProvider _storageProvider;
	    private readonly IRawTwitterCredentialsFactory _rawTwitterCredentialsFactory;

	    public MySqlTwitterStorage(
            IStorageProvider storageProvider,
            IRawTwitterCredentialsFactory rawTwitterCredentialsFactory)
	    {
	        _storageProvider = storageProvider;
	        _rawTwitterCredentialsFactory = rawTwitterCredentialsFactory;
	    }

	    public RawTwitterCredentials GetCredentials(int userId){
			var sql = "SELECT ut.UserID, ut.TwitterName, ut.Key, ut.Secret FROM usertwitter ut WHERE ut.UserID = {0}";
	        sql = string.Format(sql, userId);
			return GetCredentialsFromSql(sql);
		}

		public int AddCredentials(int userId, RawTwitterCredentials credentials){
			var sql = "INSERT INTO usertwitter (UserID, TwitterName, `Key`, Secret) VALUES ({0}, '{1}', '{2}')";
			sql = string.Format(sql, userId, credentials.Key, credentials.Secret);
			return _storageProvider.ExecuteInsert(sql);
		}

		public bool ClearCredentials(int userId){
			var sql = "DELETE FROM usertwitter WHERE UserID = {0}";
			sql = string.Format(sql, userId);
            var rowCount = _storageProvider.Execute(sql);
			return rowCount > 0;
		}

		private RawTwitterCredentials GetCredentialsFromSql(string sql){
			var reader = _storageProvider.Query(sql);
			if(reader.Read()){
				return _rawTwitterCredentialsFactory.Create(reader);
			}
			return null;
		}

	}

}